src/librustc/middle/ty.rs

```diff
@@ -1753,9 +1753,11 @@ pub struct TypeParameterDef<'tcx> {
     pub def_id: ast::DefId,
     pub space: subst::ParamSpace,
     pub index: u32,
-    pub bounds: ParamBounds<'tcx>,
     pub default: Option<Ty<'tcx>>,
     pub object_lifetime_default: Option<ObjectLifetimeDefault>,
+
+    // FIXME #22436 -- once #22436 lands, we can remove this
+    pub region_bounds: Vec<ty::Region>,
```

Comment:
It looks like #22436 has landed, could we remove this before landing this patch? if you don't have time I would be happy to help.